aws_common/test/sdk_utils/throttling_manager_test.cpp

```diff
@@ -109,19 +109,21 @@ void MakeCalls(ThrottledClient * throttled_client, std::chrono::milliseconds dur
  */
 TEST(ThrottlingManagerTest, multiThreadedClientThrottling)
 {
-  int milliseconds_to_run = 3500, sleep_duration_in_us = 150, max_tps = 125;
+  float milliseconds_to_run = 3500, max_tps = 125;
+  int sleep_duration_in_us = 150;
   int threads_to_spawn = std::max((unsigned int)2, std::thread::hardware_concurrency());
   ThrottledClient throttled_client(max_tps);
   std::vector<std::thread> threads;
   for (int tid = 0; tid < threads_to_spawn; tid++) {
     threads.push_back(std::thread(MakeCalls, &throttled_client,
-                                  std::chrono::milliseconds(milliseconds_to_run),
+                                  std::chrono::milliseconds(static_cast<int>(milliseconds_to_run)),
                                   std::chrono::microseconds(sleep_duration_in_us)));
   }
   for (auto & t : threads) {
     t.join();
   }
-  int expected_non_throttled_call_count = max_tps * milliseconds_to_run / 1000;
+  int expected_non_throttled_call_count = static_cast<int>(
+    std::ceil(max_tps * milliseconds_to_run / 1000));
```

Comment:
`std::ceil(max_tps * milliseconds_to_run / (float) 1000))` will solve your issue, or `1000.0`

Reply:
Everything here (`milliseconds_to_run`, `sleep_duration_in_us`, `max_tps`, and consequently `expected_non_throttled_call_count`) seems to be deterministic, so they can all just even be `constexpr`s.

The old calculation gives 437.5 truncated to 437, and the new calculation also gives 437.5 but ceiled to 438. My concern is this is a lot of changes just to get `expected_non_throttled_call_count` to be different by 1. And why does `expected_non_throttled_call_count` being different by just 1 solve the problem? It doesn't feel like a robust solution.

Reply:
changed to `std::ceil(max_tps * milliseconds_to_run / (float) 1000))`.

Because we always want to round up. We never caught this before release because I think it never reached that value.